src/main/java/com/yelp/nrtsearch/server/luceneserver/IndexState.java

```diff
@@ -226,6 +227,15 @@ public ThreadPoolExecutor getSearchThreadPoolExecutor() {
     return searchThreadPoolExecutor;
   }
 
+  public IdFieldDef getIdFieldDef() {
+    for (Map.Entry<String, FieldDef> entry : fields.entrySet()) {
+      if (entry.getValue() instanceof IdFieldDef) {
+        return (IdFieldDef) entry.getValue();
+      }
+    }
```

Comment:
It might be nice to avoid this repeated field searching. Could maybe have IndexState.addField() store the IdFieldDef when it is added?